app/scripts/views/mixins/flow-begin-mixin.js

```diff
@@ -7,21 +7,27 @@
 define(function (require, exports, module) {
   'use strict';
 
-  var $ = require('jquery');
-  var AuthErrors = require('lib/auth-errors');
+  var Flow = require('models/flow');
 
   module.exports = {
     afterRender: function () {
-      var flowId = this.user.get('flowId');
-      var flowBeginTime = parseInt($('body').attr('data-flow-begin'), 10);
+      var self = this;
 
-      if (! flowBeginTime) {
-        flowBeginTime = undefined;
-        this.logError(AuthErrors.toError('INVALID_DATA_FLOW_BEGIN_ATTR'));
-      }
+      self.flow = new Flow({
+        sentryMetrics: self.sentryMetrics,
+        window: self.window
+      });
 
-      this.metrics.setActivityEventMetadata('flowBeginTime', flowBeginTime);
-      this.metrics.logFlowBegin(flowId, flowBeginTime);
+      return self.flow.fetch().then(function () {
```

Comment:
This is the offending line. We now have two `afterRender` functions that return a promise. This and [afterRender in sign_up.js](https://github.com/mozilla/fxa-content-server/blob/d45a60cc79aad37af1a64eadd07b695bcc29078c/app/scripts/views/sign_up.js#L101). Whichever resolves first will cause [displayStatusMessages](https://github.com/mozilla/fxa-content-server/blob/d45a60cc79aad37af1a64eadd07b695bcc29078c/app/scripts/views/base.js#L199) to be called. Since the afterRender in sign_up.js itself returns a promise, `displayStatusMessages` is called, and THEN [FormView's afterRender](https://github.com/mozilla/fxa-content-server/blob/d45a60cc79aad37af1a64eadd07b695bcc29078c/app/scripts/views/sign_up.js#L111) function is called, which calls `enableSubmitIfValid`. Since the error message was previously displayed, the model contains no `error` and the error message is taken away.